tests/influence/test_influences.py

```diff
@@ -228,58 +267,151 @@ def test_influences_nn(
+        model,
+        training_data=train_data_loader,
+        test_data=test_data_loader,
+        progress=True,
+        influence_type=influence_type,
+        inversion_method=InversionMethod.Direct,
+        hessian_regularization=hessian_reg,
     )
-    multiple_influences = {}
-    for inversion_method in InversionMethod:
-        influences = compute_influences(
-            TorchTwiceDifferentiable(nn_architecture, loss),
-            training_data=train_data_loader,
-            test_data=test_data_loader,
-            progress=True,
-            influence_type=influence_type,
-            inversion_method=inversion_method,
-            hessian_regularization=hessian_reg,
-            **inversion_method_kwargs[inversion_method],
-        ).numpy()
-        assert not np.any(np.isnan(influences))
-        multiple_influences[inversion_method] = influences
-
-    for infl_type, influences in multiple_influences.items():
-        if infl_type == "direct":
-            continue
-        assert np.allclose(
-            influences,
-            multiple_influences["direct"],
-            rtol=1e-1,
-        ), f"Failed method {infl_type}"
-        if influence_type == InfluenceType.Up:
-            assert influences.shape == (test_data_len, data_len)
-        elif influence_type == InfluenceType.Perturbation:
-            assert influences.shape == (test_data_len, data_len, *input_dim)
+
+    approx_influences = compute_influences(
+        model,
+        training_data=train_data_loader,
+        test_data=test_data_loader,
+        progress=True,
+        influence_type=influence_type,
+        inversion_method=inversion_method,
+        hessian_regularization=hessian_reg,
+        **inversion_method_kwargs,
+    ).numpy()
+    assert not np.any(np.isnan(approx_influences))
+
+    assert np.allclose(approx_influences, direct_influence, rtol=1e-1)
+
+    if influence_type == InfluenceType.Up:
+        assert approx_influences.shape == (test_data_len, data_len)
+
+    if influence_type == InfluenceType.Perturbation:
+        assert approx_influences.shape == (test_data_len, data_len, *input_dim)
+
     # check that influences are not all constant
-    assert not np.all(influences == influences.item(0))
+    assert not np.all(approx_influences == approx_influences.item(0))
+
+
+def minimal_training(
+    model: torch.nn.Module,
+    dataloader: DataLoader,
+    loss_function: torch.nn.modules.loss._Loss,
+    lr=0.01,
+    epochs=50,
+):
+    """Trains a PyTorch model using L-BFGS optimizer.
+
+    Args:
+        model (Module): The PyTorch model to be trained.
+        dataloader (DataLoader): DataLoader providing the training data.
+        loss_function (Module): The loss function to be used for training.
+        lr (float, optional): The learning rate for the L-BFGS optimizer. Defaults to 0.01.
+        epochs (int, optional): The number of training epochs. Defaults to 50.
+
+    Returns:
+        Module: The trained model.
```

Comment:
Please make sure to use a consistent style for the docstrings.

Reply:
Done